src/Athena/Controllers/api/CourseController.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Athena.Core.Repositories;
+using Athena.Core.Models;
+using Athena.Exceptions;
+using System.Net;
+using Athena.Extensions;
+
+namespace Athena.Controllers.api
+{
+    [Route("api/v1/[Controller]")]
+    public class CourseController : Controller
+    {
+        private readonly ICourseRepository _courses;
+        private readonly IInstitutionRepository _institutions;
+        private readonly IStudentRepository _students;
+        private readonly IRequirementRepository _requirements;
+
+        public CourseController(ICourseRepository coursesRepository, IInstitutionRepository institutionRepository, IStudentRepository studentRepository, IRequirementRepository requirementRepository)
+        {
+            _courses = coursesRepository ?? throw new ArgumentNullException(nameof(coursesRepository));
+            _institutions = institutionRepository ?? throw new ArgumentNullException(nameof(institutionRepository));
+            _students = studentRepository ?? throw new ArgumentNullException(nameof(studentRepository));
```

Comment:
I don't see this used anywhere in this controller. Please remove it if it's not used.